Include/pystate.h

```diff
@@ -14,13 +14,20 @@ removed (with effort). */
 /* Forward declarations for PyFrameObject, PyThreadState
    and PyInterpreterState */
 struct _ts;
+struct _tsr;
 struct _is;
 
 /* struct _ts is defined in cpython/pystate.h */
 typedef struct _ts PyThreadState;
+/* struct _tsr is defined in cpython/pystate.h */
+typedef struct _tsr PyTStateRef;
 /* struct _is is defined in internal/pycore_interp.h */
 typedef struct _is PyInterpreterState;
 
+PyAPI_DATA(PyTypeObject) PyTStateRef_Type;
+
+#define PyTStateRef_Check(op) Py_IS_TYPE(op, &PyTStateRef_Type)
```

Comment:
I've put this here so that `PyTStateRef_Check(op)` could be used in `_testcapi`. But given how simple the definition is I could get rid of it.

Reply:
@JelleZijlstra I can't get this to work if I remove the `PyAPI_DATA` export. Any idea where I could put that to avoid affecting the stable API? I think one thing to consider, though, is that the new `PyTStateRef` objects are surfaced to Python.

Reply:
I'm not too experienced with the C API, but I am pretty confident we don't want to expose this class in the C API without a strong reason. Generally we're trying to restrict the C API so that it is easier to evolve CPython. For example, changes to the PyThreadState have caused trouble several times. https://devguide.python.org/c-api/ says internal symbols should ideally be in Include/internal.

https://docs.python.org/3/c-api/init.html#c.PyThreadState also says that only the "interp" member of this struct is public, so putting it here isn't enough to allow well-behaved external users of the API to access the thread name. I think you should prefix the new struct members with underscores and add a new, documented and public function `PyThreadState_GetName()` for extensions to access the thread name.